dev/PushNotifications/PushNotificationsLongRunningTask/NotificationListenerManager.cpp

```diff
@@ -15,30 +18,31 @@ void NotificationListenerManager::Initialize(std::shared_ptr<ForegroundSinkManag
 
 void NotificationListenerManager::AddListener(std::wstring appId, std::wstring processName)
 {
-    auto lock = m_lock.lock_exclusive();
-
     THROW_HR_IF(E_INVALIDARG, appId.empty());
     THROW_HR_IF(E_INVALIDARG, processName.empty());
 
-    if (m_notificationListeners.find(appId) == std::end(m_notificationListeners))
```

Comment:
Personally, I would just revert the old code and move the lock down. I don't see why we need any modifications to this method.

Reply:
Please looked at the scenario described in my other comment.